R/data_to_wide.R

```diff
@@ -4,11 +4,18 @@
 #' the number of rows. This is a dependency-free base-R equivalent of
 #' `tidyr::pivot_wider()`.
 #'
-#' @param data A data frame to pivot.
-#' @param id_cols The name of the column that identifies the rows. If `NULL`,
-#' it will use all the unique rows.
-#' @param names_from The name of the column that contains the levels to be
-#' used as future column names.
+#' @param data A data frame to convert to wide format, so that it has more
+#' columns and fewer rows post-widening than pre-widening.
+#' @param by The name of the column that identifies the rows in the data by
```

Comment:
I personally prefer `by`, both because it's in line with renaming the other arguments, especially since `by` referes to a "grouping" variable. Furthermore, `by` makes clearer that this specified variables is used for gathering those rows that are spread as new columns.
However, since we already allow `select` (easystats-name) and `cols` (tidyverse-name), we could probably use both names here, too?

Reply:
I move this to the main thread so that we don't have arguments split between the main thread and review comments